### PR TITLE
Moved form components to Twig namespace

### PR DESCRIPTION
**Pull request by core23**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Added `Twig\` subnamespace for all twig components

### Changed
 - Deprecated `Sonata\CoreBundle\Component\Status` class
 - Deprecated `Sonata\CoreBundle\FlashMessage` classes
 - Deprecated `Sonata\CoreBundle\Twig\Extension` classes
 - Deprecated `Sonata\CoreBundle\Twig\Node` classes
 - Deprecated `Sonata\CoreBundle\Twig\TokenParser` classes

```

## To do

- [x] Update the documentation
- [x] Add an upgrade note

## Subject

This is the second PR for the repo split #579

### CODE REVIEW

File: src/CoreBundle/Component/Status/StatusClassRendererInterface.php [hunk 1]
```diff
@@ -11,29 +11,17 @@
 
 namespace Sonata\CoreBundle\Component\Status;
 
+@trigger_error(
+    'The '.__NAMESPACE__.'\FlashManager class is deprecated since version 3.x and will be removed in 4.0.'
+    .' Use Sonata\Twig\FlashMessage\FlashManager instead.',
```
Comment (kunicmarko20) [inline src/CoreBundle/Component/Status/StatusClassRendererInterface.php]: `FlashManager` and you are in `StatusClassRendererInterface`

File: src/CoreBundle/Twig/Node/DeprecatedTemplateNode.php [hunk 2]
```diff
@@ -11,29 +11,17 @@
 
 namespace Sonata\CoreBundle\Twig\Node;
 
-use Twig\Compiler;
-use Twig\Node\Expression\AbstractExpression;
-use Twig\Node\Node;
+@trigger_error(
+    'The '.__NAMESPACE__.'\TemplateBoxNode class is deprecated since version 3.x and will be removed in 4.0.'
+    .' Use Sonata\Twig\Node\TemplateBoxNode\FlashManager instead.',
```
Comment (kunicmarko20) [inline src/CoreBundle/Twig/Node/DeprecatedTemplateNode.php]: FlashManager again, also you mention `TemplateBoxNode` while in extends there isn't that part of the namespace

File: src/CoreBundle/Twig/Node/TemplateBoxNode.php [hunk 3]
```diff
@@ -11,81 +11,15 @@
...
 
-use Symfony\Component\Translation\TranslatorInterface;
-use Twig\Compiler;
-use Twig\Node\Expression\AbstractExpression;
-use Twig\Node\Node;
-
-class TemplateBoxNode extends Node
+@trigger_error(
+    'The '.__NAMESPACE__.'\TemplateBoxNode class is deprecated since version 3.x and will be removed in 4.0.'
+    .' Use Sonata\Twig\TokenParser\TemplateBoxNode instead.',
```
Comment (kunicmarko20) [inline src/CoreBundle/Twig/Node/TemplateBoxNode.php]: Again different in the `extends`

File: src/Twig/Extension/DeprecatedTemplateExtension.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+use Sonata\Twig\TokenParser\DeprecatedTemplateTokenParser;
+use Twig\Extension\AbstractExtension;
+
+/**
+ * @author Marko Kunic <kunicmarko20@gmail.com>
+ */
+class DeprecatedTemplateExtension extends AbstractExtension
+{
+    /**
+     * {@inheritdoc}
```
Comment (kunicmarko20) [inline src/Twig/Extension/DeprecatedTemplateExtension.php]: do we need this?

**Comment by greg0ire** [comment]
Why did coverage decrease so much :thinking:

**Comment by core23** [comment]
> Why did coverage decrease so much? 🤔

Don't know. Did we remove coverage? I can't see the report anymore

**Comment by greg0ire** [comment]
I don't know why coveralls only works "sometimes". For the decrease, could it have to do with ` tests/CoreBundle/Twig/Extension/StatusExtensionTest.php`?

**Comment by core23** [comment]
Maybe the decrease is because ofthe `deprecation` warnings and the `legacy` tags?

File: tests/CoreBundle/Twig/Extension/StatusExtensionTest.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        $this->assertContainsOnlyInstancesOf('Twig_SimpleFilter', $filters);
+    }
+
+    /**
+     * @group legacy
+     */
+    public function testStatusClassDefaultValue()
+    {
+        $extension = new StatusExtension();
+        $statusService = $this->getMockBuilder(StatusClassRendererInterface::class)
```
Comment (greg0ire) [inline tests/CoreBundle/Twig/Extension/StatusExtensionTest.php]: Please use `createMock` directly.

Comment (greg0ire) [reply tests/CoreBundle/Twig/Extension/StatusExtensionTest.php]: Oh sorry, forgot this was a move, disregard.

**Comment by greg0ire** [comment]
> Maybe the decrease is because of the deprecation warnings and the legacy tags?

Ah it probably has to do with deprecations indeed: new code was introduced to deprecate the old class names right? So this must be it.

**Comment by greg0ire** [comment]
Thanks @core23 !

**Comment by ossinkine** [comment]
Why does services still point to deprecated classes?

**Comment by core23** [comment]
Because we have to stay BC

**Comment by ossinkine** [comment]
@core23 There is no BC break if we change service classes to new ones

**Comment by ossinkine** [comment]
Was wrong. Changing classes really breaks BC. Perhaps it makes sense to do the same as #629.
Anyway, deprecation errors are thrown now. How to avoid them?

**Pull request merged**